src/Render/RelativeFile.cs

```diff
@@ -43,15 +43,10 @@ internal sealed class RelativeFile {
 		}
 
 		private string GetDestinationPath( string newName = null ) {
-			string path;
-			string relpath = System.IO.Path.GetRelativePath( SourceRoot, Path );
-			if ( newName != null ) {
-				string relpathWithoutFilename = Directory.GetParent( relpath ).FullName;
-				path = System.IO.Path.Join( DestinationRoot, relpathWithoutFilename, newName );
-			} else {
-				path = System.IO.Path.Join( DestinationRoot, relpath );
-			}
-			return path;
+			newName = newName ?? System.IO.Path.GetFileName( Path );

```

Comment:
Cleaned this function up a bit. In particular, `Directory.GetParent().FullName` was giving an absolute path, causing some weird directory structure.